walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -40,29 +47,57 @@
     finalTimes = new ArrayList<>(finalTimes);
 
     personRelevantEvents.sort(Event.ORDER_BY_START);
-    
+    combinedRelevantEvents.sort(Event.ORDER_BY_START);
+
     // remove events that are contained within others
-    List<Event> toDelete = new ArrayList<Event>();
-    for (int i = 0; i < personRelevantEvents.size() - 1; i++) {
-      if (personRelevantEvents.get(i).getWhen().contains(personRelevantEvents.get(i+1).getWhen())) {
-        toDelete.add(personRelevantEvents.get(i+1));
+    List<Event> toDeletePerson = new ArrayList<Event>();
+    for (int i = 0; i < personRelevantEvents.size(); i++) {
+      for (int o = i + 1; o < personRelevantEvents.size(); o++) {
+        if (personRelevantEvents.get(i).getWhen().contains(personRelevantEvents.get(o).getWhen())) {
+          toDeletePerson.add(personRelevantEvents.get(o));
+        }
       }
     }
-    for (Event e : toDelete) {
+    for (Event e : toDeletePerson) {
       personRelevantEvents.remove(e);
     }
+
+    List<Event> toDeleteCombined = new ArrayList<Event>();
+    for (int i = 0; i < combinedRelevantEvents.size(); i++) {
+      for (int o = i + 1; o < combinedRelevantEvents.size(); o++) {
+        if (combinedRelevantEvents.get(i).getWhen().contains(combinedRelevantEvents.get(o).getWhen())) {
+          toDeleteCombined.add(combinedRelevantEvents.get(o));
+        }
+      }
+    }
+    for (Event e : toDeleteCombined) {
+      combinedRelevantEvents.remove(e);
+    }
```

Comment:
Maybe refactor into a common method because this section is very similar to the previous section.

Reply:
Got it!